.github/workflows/test.yml

```diff
@@ -0,0 +1,28 @@
+name: Build & test
+
+on:
+  push:
+    branches: [ "main", "setup-ci" ]
```

Comment:
Should we change this to run on all branches rather than just main?

Reply:
When someone offers a PR, _from `main`_ the workflow runs. Which may be all we want?

Reply:
(I am mostly "worried about duplicate builds", but that's fine.) I _think_ this configuration will let any next push to a branch which has an open PR will execute the workflow.

Reply:
afaik, line 5 will mean that the action only runs on those branches?

Reply:
I like your idea @artismarti - sometimes you want to push a branch before you even make a PR, and it would be useful to get feedback on those commits too. I see no reason not to make this more liberal.

Reply:
Resolved in 24c0fbf0fd727fdfd240e762a563baa5013a7552